cluster-autoscaler/core/scale_down.go

```diff
@@ -146,6 +177,58 @@ func calculateCoresAndMemoryTotal(nodes []*apiv1.Node, timestamp time.Time) (int
 	return coresTotal, memoryTotal
 }
 
+func calculateGpusTotal(nodes []*apiv1.Node, cp cloudprovider.CloudProvider, timestamp time.Time) (map[string]int64, error) {
+	type gpuInfo struct {
+		name  string
+		count int64
+	}
+
+	result := make(map[string]int64)
+	ngCache := make(map[string]gpuInfo)
+	for _, node := range nodes {
+		if isNodeBeingDeleted(node, timestamp) {
+			// Nodes being deleted do not count towards total cluster resources
+			continue
+		}
+		nodeGroup, err := cp.NodeGroupForNode(node)
+		if err != nil {
+			return nil, errors.ToAutoscalerError(errors.CloudProviderError, err).AddPrefix("can not get node group for node %v when calculating cluster gpu usage", node.Name)
+		}
+		if reflect.ValueOf(nodeGroup).IsNil() {
+			// TODO[lukaszos] why do we need this? do we?
```

Comment:
Welcome to Golang: https://golang.org/doc/faq#nil_error

Reply:
I understand why we need to use reflect to check for nil, what I don't understand is why do we need this if at all? What I'm seeing is:

```
if x == a {
  x = a
}
```
Am I missing something obvious here???

Reply:
The second nil is a nil interface, so it can be safely used later

Reply:
Added comment